prov/sockets/src/sock_fabric.c

```diff
@@ -719,7 +719,7 @@ static int sock_getinfo(uint32_t version, const char *node, const char *service,
 		}
 	}
 
-	return ret;
+	return (!*info) ? ret : 0;
```

Comment:
Yes, ```ret``` will always be set to the return value of the last ```sock_node_getinfo```() call and that's what we were returning before, but if we have ```*info```, then we should return 0 as one/more interfaces were discovered by ```sock_node_getinfo```()